File: InventoryApp.Data/Models/InventoryDeliveryVoucherDetail.cs
using InventoryApp.Data.Helper;
using Microsoft.EntityFrameworkCore;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace InventoryApp.Data.Models
{
    [Table("InventoryDeliveryVoucherDetail")]
    [Index(nameof(InventoryDeliveryVoucherId), nameof(ShipmentId), IsUnique = true)]
    public class InventoryDeliveryVoucherDetail
    {
        [Key]
        public Guid Id { get; set; }
        public Guid InventoryDeliveryVoucherId { get; set; }
        public InventoryDeliveryVoucher? InventoryDeliveryVoucher { get; set; }
        public Guid ShipmentId { get; set; }
        public Shipment? Shipment { get; set; }
        public Materials Material { get; set; }
        public Guid MaterialId { get; set; }
        public int QuantityDelivery { get; set; }
        public double Price { get; set; }
    }
}

File: InventoryApp.Client/src/app/module/inventory-delivery-voucher/model/inventory-delivery-voucher-detail.ts
import { Material } from "../../material/model/material";
import { Shipment } from "../../shipment/model/shipment";

export interface InventoryDeliveryVoucherDetail{
    materialId:string;
    material: Material;
    shipmentId:string;
    shipment:Shipment;
    quantityDelivery:string;
    price:string;
    amount: string;
    priceTotalItem:number;
}